extensions/email/src/main/java/org/mule/extension/email/internal/commands/SetFlagCommand.java

```diff
@@ -52,4 +51,27 @@ public void set(MailboxConnection connection, String folderName, Flag flag, long
       throw new EmailException(format("Error while setting [%s] flag in email of id [%s]", flag.toString(), emailId), e);
     }
   }
+
+  /**
+   * Sets the specified {@code flag} to the email of mailbox number {@code number}
+   * <p>
+   * This method only works for {@link UIDFolder}s, that are handled by the IMAP protocol
+   *
+   * @param connection the associated {@link MailboxConnection}.
+   * @param folderName the name of the folder where the emails are stored.
+   * @param flag       the {@link Flag} that wants to be set in the email message.
+   * @param number     the number of the email in the corresponding {@link Folder} of name {@code folderName}
+   */
+  public void setByNumber(MailboxConnection connection, String folderName, Flag flag, int number) {
```

Comment:
is this by number or by id? This isn't relative to the message position on the mailbox isn't it?